agent/tests/test_php_wrapper.c

```diff
@@ -85,6 +90,460 @@ NR_PHP_WRAPPER(test_add_2_arrays) {
+                               test_name_txn_before_ok, NULL, "three",
+                               "one:beforeok,two:beforeok,three:beforeok");
+  /*
+   *
+   * 3) IF wrapper function is called after NR_PHP_WRAPPER_CALL or called in
+   * func_end AND NR_NOT_OK_TO_OVERWRITE is set THEN the LAST wrapped function
+   * encountered determines the txn name.
+   * All functions set the txn after NR_PHP_WRAPPER_CALL and/or in OAPI
+   * func_begin and use NR_NOT_OK_TO_OVERWRITE.
+   *
+   * Expecting `three` to name txn.
+   */
+  setup_nested_framework_calls(
+      NULL, test_name_txn_after_not_ok, NULL, test_name_txn_after_not_ok, NULL,
+      test_name_txn_after_not_ok, "three",
+      "one:afternotok,two:afternotok,three:afternotok");
+
+  /*
+   * 4) IF wrapper function is called after NR_PHP_WRAPPER_CALL or called in
+   * func_end AND NR_OK_TO_OVERWRITE is set THEN the FIRST wrapped function
+   * encountered determines the txn name.
+   *
+   * All functions set the txn before NR_PHP_WRAPPER_CALL and/or in OAPI
+   * func_begin and use NR_OK_TO_OVERWRITE.
+   *
+   * Expecting `one` to name txn.
+   */
+  setup_nested_framework_calls(
+      NULL, test_name_txn_after_ok, NULL, test_name_txn_after_ok, NULL,
+      test_name_txn_after_ok, "one", "one:afterok,two:afterok,three:afterok");
+
+  /*
+   * 5) If there are nested functions that have wrapped functions called before
+   * NR_PHP_WRAPPER_CALL or called in func_begin AND that also have called after
+   * NR_PHP_WRAPPER_CALL or called in func_end if the after call uses
+   * NR_NOT_OK_TO_OVERWRITE, then rule 1 or 2 applies depending on whether a
+   * before_func used NR_NOT_OK_TO_OVERWRITE or NR_NOT_TO_OVERWRITE.
+   *
+   * 6) If there are nested functions that have wrapped functions called before
+   * NR_PHP_WRAPPER_CALL or called in func_begin AND that also have called after
+   * NR_PHP_WRAPPER_CALL or called in func_end if the after call uses
+   * NR_OK_TO_OVERWRITE, then rule 4 applies.
+   *
+   * Basically a mix and match situation where we have some best tries at naming
+   * a nested function transaction via one txn naming wrapped function, but if
+   * something better comes along via a DIFFERENT wrapped function, we'd prefer
+   * that. Special functions are mixed with regards to calling before/after and
+   * NR_NOT_OK_TO_OVERWRITE/NR_OK_TO_OVERWRITE
+   */
+
+  /*
+   * After only mixes of NR_NOT_OK_TO_OVERWRITE/NR_NOT_OK_TO_OVERWRITE
+   */
+
+  /*
+   * special function for one is called after and uses NR_NOT_OK_TO_OVERWRITE
+   * special function for two is called after and uses NR_OK_TO_OVERWRITE
+   * special function for three is called after and uses NR_OK_TO_OVERWRITE
+   *
+   * expect txn to be named `two`
+   */
+  setup_nested_framework_calls(NULL, test_name_txn_after_not_ok, NULL,
+                               test_name_txn_after_ok, NULL,
+                               test_name_txn_after_ok, "two",
```

Comment:
So the call stack is:
```
one
  two
    three
    after_callback_for_three
      nr_txn_set_path sets txn name to 'three' because it has not been set yet
  after_callback_for_two
    nr_txn_set_path sets txn name to 'two' because called with NR_OK_TO_OVERWRITE
after_callback_for_one
  nr_txn_set_path leaves txn name as 'two' because called with NR_NOT_OK_TO_OVERWRITE
```
Then the scenario description using suggested convention would be:
`one:name_after_call:no_overwrite, two:name_after_call:overwrite, three:name_after_call:overwrite`
and naming basically happens by going backward the above list.

Reply:
please see: [857fa16](https://github.com/newrelic/newrelic-php-agent/pull/615/commits/857fa1642e98a49e0a3b46e1404af1b6a4146637)
Convention is to describe what sort of wrapper each function is calling.

Reply:
My comment with call stack was for other reviewers. Thank you for including it in the test documentation.